packages/api-plugin-products/src/mutations/addTagsToProducts.js

```diff
@@ -15,6 +15,7 @@ export default async function addTagsToProducts(context, input) {
   const { productIds, shopId, tagIds } = input;
   const { collections: { Products } } = context;
   const totalProducts = productIds.length;
+  const { appEvents } = context;
```

Comment:
You are already deconstructing context on line 16, should probably just get `appEvents` from there.